tests/parametric/test_dynamic_configuration.py

```diff
@@ -178,35 +183,40 @@ def test_capability_tracing_enabled(self, library_env, test_agent, test_library)
         test_agent.wait_for_rc_capabilities([Capabilities.APM_TRACING_ENABLED])
 
     @parametrize(
-        "library_env", [{**DEFAULT_ENVVARS}, {**DEFAULT_ENVVARS, "DD_TRACE_ENABLED": "false"},],
+        "library_env", [{**DEFAULT_ENVVARS}, {**DEFAULT_ENVVARS, "DD_TRACE_ENABLED": False},],
     )
     def test_tracing_client_tracing_enabled(self, library_env, test_agent, test_library):
-        if library_env.get("DD_TRACE_ENABLED", True):
+        trace_enabled_env = library_env.get("DD_TRACE_ENABLED", True)
+        if trace_enabled_env:
             with test_library:
-                with test_library.start_span("test"):
+                with test_library.start_span("allowed"):
                     pass
             test_agent.wait_for_num_traces(num=1, clear=True)
             assert True, (
                 "DD_TRACE_ENABLED=true and unset results in a trace being sent."
                 "wait_for_num_traces does not raise an exception."
             )
 
-        set_and_wait_rc(test_agent, config_overrides={"tracing_enabled": "false"})
+        _set_rc(test_agent, _create_rc_config({"tracing_enabled": False}))
+        # if tracing is disabled via DD_TRACE_ENABLED, the RC should not re-enable it
+        # nor should it send RemoteConfig apply state
+        if trace_enabled_env:
+            test_agent.wait_for_telemetry_event("app-client-configuration-change", clear=True)
+            test_agent.wait_for_rc_apply_state("APM_TRACING", state=2, clear=True)
```

Comment:
I think we should still expect the RC apply_state even if the app is started with DD_TRACE_ENABLED=false

Reply:
From what I understood, if DD_TRACE_ENABLED == false, it doesn't really start the tracer nor the RC client, and thus doesn't expect the apply state ?

Reply:
I don't think that behaviour is specified anywhere 💀. We'll for sure at some point want RC to be enabled even if `DD_TRACE_ENABLED` is false so that we can remotely enable other products.

Does the go library disable RC if `DD_TRACE_ENABLED` is false? We might have to special case the test for now 😢

Reply:
From discussion offline, Go lib doesn't start RC if disabled from the start